ruma-client-api/src/r0/account/register.rs

```diff
@@ -13,7 +13,7 @@ ruma_api! {
         name: "register",
         path: "/_matrix/client/r0/register",
         rate_limited: true,
-        authentication: None,
+        authentication: AccessToken,
```

Comment:
Since registering with [admin styles permissions](https://matrix.org/docs/spec/application_service/r0.1.2#server-admin-style-permissions) requires passing in the access_token additionally to the login type I'm not actually sure how to properly do that here. Maybe it needs another Request-variant?